test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,809 @@
+                    ).to.be.equal(ethers.constants.AddressZero);
+                });
+            });
+        });
+    });
+
+    describe("#stagedProtocolGovernanceTimestamp", () => {
+        it("returns timestamp after which #commitStagedProtocolGovernance can be called", async () => {
+            await this.subject
+                .connect(this.admin)
+                .stageProtocolGovernance(this.newProtocolGovernance.address);
+            expect(
+                Math.abs(
+                    Number(
+                        await this.subject.stagedProtocolGovernanceTimestamp()
+                    ) -
+                        now() -
+                        Number(await this.protocolGovernance.governanceDelay())
+                )
+            ).to.be.lessThanOrEqual(1);
+        });
+
+        describe("access control:", () => {
+            it("allowed: any address", async () => {
+                await withSigner(randomAddress(), async (s) => {
+                    await expect(
+                        this.subject.stagedProtocolGovernanceTimestamp()
+                    ).to.not.be.reverted;
+                });
+            });
+        });
+
+        describe("edge cases", () => {
+            describe("when nothing is staged", () => {
+                it("returns 0", async () => {
+                    expect(
+                        await this.subject.stagedProtocolGovernanceTimestamp()
+                    ).to.be.equal(0);
+                });
+            });
+            describe("right after #commitStagedProtocolGovernance was called", () => {
+                it("returns 0", async () => {
+                    await this.subject
+                        .connect(this.admin)
+                        .stageProtocolGovernance(
+                            this.newProtocolGovernance.address
+                        );
+                    await sleep(
+                        Number(await this.protocolGovernance.governanceDelay())
+                    );
+                    await this.subject
+                        .connect(this.admin)
+                        .commitStagedProtocolGovernance();
+                    expect(
+                        await this.subject.stagedProtocolGovernanceTimestamp()
+                    ).to.be.equal(0);
+                });
+            });
+        });
+    });
+
+    describe("#vaultsCount", () => {
+        it("returns the number of registered vaults", async () => {
+            expect(await this.subject.vaultsCount()).to.be.equal(2);
+        });
+
+        describe("access control:", () => {
+            it("allowed: any address", async () => {
+                await withSigner(randomAddress(), async (s) => {
+                    await expect(this.subject.vaultsCount()).to.not.be.reverted;
+                });
+            });
+        });
+        describe("edge cases", () => {
+            describe("when new vault is registered", () => {
```

Comment:
Instead of having this test it'd be nice to have a proptest: #vaultsCount() == #vaults().length. Check with @MihanixA how it's done in ProtocolGovernace tests